appeals/api/src/server/repositories/representation.repository.js

```diff
@@ -61,25 +61,33 @@ export const getFinalCommentsByAppealId = (appealId) => {
 };
 
 /**
- *
  * @param {number} appealId
  * @param {number} pageNumber
  * @param {number} pageSize
  * @param {string|undefined} status
- * @returns {Promise<import('@pins/appeals.api').Schema.Representation[]>}
+ * @returns {Promise<{ itemCount: number, comments: import('@pins/appeals.api').Schema.Representation[] }>}
  */
-export const getThirdPartyCommentsByAppealId = (
+export const getThirdPartyCommentsByAppealId = async (
 	appealId,
 	pageNumber = 0,
 	pageSize = 30,
 	status
 ) => {
-	return databaseConnector.representation.findMany({
-		where: {
-			appealId,
-			representationType: APPEAL_REPRESENTATION_TYPE.COMMENT,
-			...(status && { status })
-		},
+	// Define the where clause for filtering
+	const whereClause = {
+		appealId,
+		representationType: APPEAL_REPRESENTATION_TYPE.COMMENT,
+		...(status && { status })
+	};
+
+	// Fetch the total count of matching representations
+	const itemCount = await databaseConnector.representation.count({
```

Comment:
This is good, but a better way is to make a single roundtrip to the DB, so you can wrap in a transaction, e.g.:

```
const transaction = [
  databaseConnector.representation.count({...}),
  databaseConnector.representation.findMany({...})
];

return databaseConnector.$transaction(transaction)
```

Reply:
sorted